src/llm_vm/onsite_llm.py

```diff
@@ -527,3 +485,56 @@ def finetune(self, dataset, optimizer, c_id, small_model_filename=None):
         # optimizer.storage.set_training_in_progress(c_id, False)
         # if old_model is not None:
         #     openai.Model.delete(old_model)
+
+
+class BaseCtransformers(BaseOnsiteLLM):
+    """
+    Base Class for running Ctransformers/GGML models
+
+    Attributes:
```

Comment:
Do we need to abc the model file name property? It’ll be set like half the time with the most common models.  For uri I assume the abc stuff in the base class handles that or do we need to keep it abstract? Are we wanting this class to be instantiated directly or be an abstract class we sub class ?

Reply:
i intended it to be an abstract subclass that handles most of the repeated logic in the Ctransformer models

Reply:
from this it should be easy to add the rest of the models in the same way as the quantized llama